<launch>
    <arg name="camera0" default="frontleft" />
    <arg name="camera1" default="frontright" />

    <group ns="rtabmap">
        <!-- Use RGBD synchronization -->
        <!-- Here is a general example using a standalone nodelet, 
         but it is recommended to attach this nodelet to nodelet 
         manager of the camera to avoid topic serialization -->
        <!-- RGBD sync -->
        <include file="$(find spot_rtabmap)/launch/include/rgbd_sync_dual.launch">
            <arg name="camera0" value="$(arg camera0)" />
            <arg name="camera1" value="$(arg camera1)" />
        </include>


        <node name="rtabmap" pkg="rtabmap_slam" type="rtabmap" output="screen"
            args="--delete_db_on_start">
            <param name="frame_id" type="string" value="body" />

            <param name="subscribe_depth" type="bool" value="false" />
            <param name="subscribe_stereo" type="bool" value="false" />
            <param name="subscribe_rgbd" type="bool" value="true" />
            <param name="subscribe_scan" type="bool" value="false" />
            <param name="subscribe_scan_cloud" type="bool" value="true" />

            <!-- rgbd image -->
            <param name="rgbd_cameras" type="int" value="2" />
            <remap from="rgbd_image0" to="$(arg camera0)_rgbd_image" />
            <remap from="rgbd_image1" to="$(arg camera1)_rgbd_image" />

            <!-- remapping -->
            <remap from="odom" to="/spot/odometry" />
            <remap from="scan_cloud" to="/spot/lidar/points" />

            <!-- params -->
            <param name="queue_size" type="int" value="50" />
            <param name="approx_sync" type="bool" value="true" />

            <!-- RTAB-Map's parameters, taken from kaiyu -->
            <param name="RGBD/ProximityBySpace" type="string" value="false" />
            <param name="RGBD/AngularUpdate" type="string" value="0.01" />
            <param name="RGBD/LinearUpdate" type="string" value="0.01" />
            <param name="RGBD/OptimizeFromGraphEnd" type="string" value="false" />
            <param name="Reg/Force3DoF" type="string" value="true" />
            <param name="Vis/MinInliers" type="string" value="12" />
            <param name="Vis/MinInliers" type="string" value="12" />
            <param name="Grid/CellSize" type="string" value="0.02" />
        </node>
    </group>
</launch>